### ISSUE TITLE
SoftMax doesn't calculate correct values for 3D inputs

### ISSUE THREAD

**Issue by hkvision** [intel-analytics/BigDL#1934]
Seems that SoftMax is not working properly for 3D inputs
```
>>> layer = SoftMax()
creating: createSoftMax
>>> input = np.random.random([1, 2, 3])
>>> input
array([[[ 0.39775828,  0.11849257,  0.32446772],
        [ 0.13931134,  0.99062448,  0.67649278]]])
>>> layer.forward(input)
array([[[ 1.,  1.,  1.],
        [ 1.,  1.,  1.]]], dtype=float32)

>>> input_data = np.array([[[1, 2, 3], [4, 5, 6]], [[7, 8, 9], [10, 11, 12]]])
>>> layer.forward(input_data)
array([[[ 0.00247262,  0.00247262,  0.00247262],
        [ 0.00247262,  0.00247262,  0.00247262]],

       [[ 0.99752742,  0.99752742,  0.99752742],
        [ 0.99752742,  0.99752742,  0.99752742]]], dtype=float32)
```

**Comment by qiuxin2012** [comment]
SoftMax's implement follows torch, it's same result with torch.

**Comment by hkvision** [comment]
For 3D input, the result doesn't match with keras for the axis.

**Comment by hkvision** [comment]
Now use transpose first before applying softmax for 3D input for keras model loading so that the result is the same with keras.
